### PR TITLE
`Development`: Remove the unused Ace code editor dependency

### PR DESCRIPTION
**Pull request by pzdr7**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

As of the PR #9230, the Ace editor is completely unused in Artemis. It can therefore be removed from the project.

### Description

- Removed the `brace` and `ace-builds` dependencies
- Removed the `AceEditorComponent` and `MarkdownEditorComponent` that relied on Ace
- Removed the `Command` hierarchy and its test cases
- Removed references to Ace in comments and other variables

### Steps for Testing

Code review. If you want, deploy to a test server and check that the markdown editors don't break.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
- [x] Code Review 3
- [x] Code Review 4

### Test Coverage

Unchanged. Global coverage changes as follows (refer to the coverage report for the latest values):

| Coverage Metric | Before  | After   |
|-----------------|---------|---------|
| Statements      | 87.29%   | 87.44%  |
| Branches        | 73.47%  | 73.53%  |
| Functions       | 81.63%  | 81.92%  |
| Lines           | 87.36%  | 87.51%  |

### Screenshots

No UI changes

## Summary by CodeRabbit

- **New Features**
	- Transitioned from Ace Editor to Monaco Editor across various components and modules, enhancing code editing capabilities.
	- Introduced new `MarkdownEditorHeight` enum for flexible height configurations in the Markdown editor.
	- Enhanced `MonacoTaskAction` class with standardized identifiers and improved regex parsing for task management.
	- Updated cursor representation in `CodeEditorMonacoComponent` for better clarity and consistency.

- **Bug Fixes**
	- Updated various components to ensure consistent functionality with the Monaco editor, improving user experience.

- **Documentation**
	- Revised comments and documentation to reflect the transition from Ace Editor to Monaco Editor, ensuring clarity for developers.

- **Chores**
	- Removed dependencies related to Ace Editor from imports and modules to streamline the codebase.

### CODE REVIEW

File: src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss
Comment (pzdr7) [inline src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss:1]: I copied the styles from the old `markdown-editor.component.scss`

File: src/main/webapp/app/shared/monaco-editor/model/monaco-editor-overlay-widget.model.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class MonacoEditorOverlayWidget extends MonacoCodeEditorElement implement
     constructor(editor: monaco.editor.ICodeEditor, id: string, domNode: HTMLElement, position: OverlayWidgetPosition) {
         super(editor, id);
         this.domNode = domNode;
-        // At the moment, the inline feedback nodes will only reach their maximum width with the following line. This workaround can be removed as soon as the Ace editor has been replaced.
+        // Ensure that the widget reaches its maximum width.
```
Comment (pzdr7) [inline src/main/webapp/app/shared/monaco-editor/model/monaco-editor-overlay-widget.model.ts:23]: It turned out that this was necessary - I guess Ace did this automatically

File: src/main/webapp/app/exercises/programming/shared/code-editor/monaco/code-editor-monaco.component.ts [hunk 2]
```diff
@@ -329,7 +329,7 @@ export class CodeEditorMonacoComponent implements OnChanges {
         }
         // In the future, there may be more than one feedback node per line.
         const feedbackNode = this.getInlineFeedbackNodeOrElseThrow(line);
-        // The lines are 0-based for Ace, but 1-based for Monaco -> increase by 1 to ensure it works in both editors.
+        // Feedback is stored with 0-based lines, but the lines of Monaco are 1-based. We add 1 to correct this
```
Comment (JohannesWt) [inline src/main/webapp/app/exercises/programming/shared/code-editor/monaco/code-editor-monaco.component.ts:332]: Nit pick to ensure everybody understands the comment
```suggestion
        // Feedback is stored with 0-based lines, but the lines of the Monaco editor used in Artemis are 1-based. We add 1 to correct this
```

File: src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss [hunk 3]
```diff
@@ -1,8 +1,32 @@
 .markdown-editor-wrapper {
+    display: flex;
+    flex-flow: column nowrap;
+    flex: 1 1 auto;
+    height: inherit;
+    color: var(--md-wrapper-color);
     background-color: var(--md-editor-background);
+
+    // This is a fix for using the fullscreen mode on Safari.
```
Comment (JohannesWt) [inline src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss:9]: Fix for what? Is there a bug in the markdown editor?

**Review by JohannesWt — COMMENTED** [review]
Code looks good just a small nitpick and one question :)

File: src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss (hunk 3 above)
Comment (pzdr7) [reply src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.scss:9]: I copied this style from the original markdown editor's style file (which the new md editor is using). I didn't want to break the Fullscreen mode of Safari, so I didn't remove the style

**Review by JohannesStoehr — APPROVED** [review]
Code removal looks good to me

**Review by JohannesWt — APPROVED** [review]
Code Looks good 👍

**Review by SimonEntholzer — APPROVED** [review]
Code looks good and client tests pass :+1: That's really a lot of removed code :rocket:

**Review by Jan-Thurner — APPROVED** [review]
Code LGTM

**Pull request merged**